backend-rust/src/indexer.rs

```diff
@@ -2053,6 +2080,37 @@ impl PreparedBakerEvent {
                 .execute(tx.as_mut())
                 .await?;
             }
+            PreparedBakerEvent::Suspended {
+                baker_id,
+            } => {
+                sqlx::query!(
+                    "UPDATE bakers
+                     SET
+                         self_suspended = $2,
+                         inactive_suspended = NULL,
+                         primed_for_suspension = NULL
+                     WHERE id=$1",
+                    baker_id,
+                    transaction_index
+                )
+                .execute(tx.as_mut())
+                .await?;
+            }
+            PreparedBakerEvent::Resumed {
+                baker_id,
+            } => {
+                sqlx::query!(
+                    "UPDATE bakers
+                     SET
+                         self_suspended = NULL,
+                         inactive_suspended = NULL,
+                         primed_for_suspension = NULL
+                     WHERE id=$1",
+                    baker_id
+                )
```

Comment:
Self-suspending and resuming shouldn't affect the `primed_for_suspension` status. The node itself only resets the primed status when a validator shows activity. Because the status only applies to current validators (i.e. in the committee, as opposed to active ones, which just have validator records on the accounts) when a snapshot is taken, this effectively clears the primed status, because the validator will now be suspended. I think the short of this is that you should just clear the `primed_for_suspension` when the validator is automatically suspended, and not in these cases.